core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -2209,6 +2209,14 @@ private void processTxFromBestChain(Transaction tx, boolean forceAddToPool) thro
         if (!doubleSpendTxns.isEmpty()) {
             // no need to addTransactionsDependingOn(doubleSpendTxns) because killTxns() already kills dependencies;
             killTxns(doubleSpendTxns, tx);
+
+            // disconnect irrelevant inputs (otherwise might cause serialization issue)
+            for(TransactionInput input : tx.getInputs()) {
```

Comment:
Yeah, I'll just change it to only diconnect when the transaction is not going to get saved anyways (doesn't send/give money and is not force added).